### PR TITLE
[DNM] *: split meta request to store/region/config

### PR DESCRIPTION
**Pull request by tiancaiamao**

### CODE REVIEW

**Comment by siddontang** [comment]
LGTM

File: pd-client/rpc_worker.go [hunk 1]
```diff
@@ -202,16 +238,65 @@ func (w *rpcWorker) getTSFromRemote(conn *bufio.ReadWriter, n int) ([]*pdpb.Time
...
+	return rsp.GetGetRegion(), nil
+}
+
+func (w *rpcWorker) getClusterConfigFromRemote(conn *bufio.ReadWriter, clusterConfigReq *pdpb.GetClusterConfigRequest) (*pdpb.GetClusterConfigResponse, error) {
+	req := pdpb.Request{
+		Header: &pdpb.RequestHeader{
+			Uuid:      uuid.NewV4().Bytes(),
+			ClusterId: proto.Uint64(w.clusterID),
+		},
+		CmdType:          pdpb.CommandType_GetRegion.Enum(),
```
Comment (qiuyesuifeng) [inline pd-client/rpc_worker.go]: Should this type be `** CommandType_GetClusterConfig`?

File: pd-client/rpc_worker.go [hunk 2]
```diff
@@ -202,16 +238,65 @@ func (w *rpcWorker) getTSFromRemote(conn *bufio.ReadWriter, n int) ([]*pdpb.Time
...
+		GetClusterConfig: clusterConfigReq,
+	}
+	rsp, err := w.do(conn, &req)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	if rsp.GetGetClusterConfig() == nil {
+		return nil, errors.New("[pd] GetRegion field in rpc response not set")
```
Comment (qiuyesuifeng) [inline pd-client/rpc_worker.go]: `GetClusterConfig` field

**Comment by qiuyesuifeng** [comment]
@tiancaiamao  Please resolve the conflict.

File: pd-client/rpc_worker.go (hunk 1 above)
Comment (qiuyesuifeng) [inline pd-client/rpc_worker.go]: @tiancaiamao  How about this?

Comment (tiancaiamao) [inline pd-client/rpc_worker.go]: a stupid mistake. 
thanks for point out.

**Comment by siddontang** [comment]
PTAL @disksing

File: server/command.go [hunk 3]
```diff
@@ -98,84 +98,115 @@ func (c *conn) getRaftCluster() (*raftCluster, error) {
...
 	}
 
-	resp := &pdpb.GetMetaResponse{
-		MetaType: request.MetaType,
+	storeID := request.GetStoreId()
+	store, err := cluster.GetStore(storeID)
+	if err != nil {
+		return nil, errors.Trace(err)
 	}
+	// Store may be nil, should we return an error instead of none result?
```
Comment (disksing) [inline server/command.go]: I think `cluster.GetStore()` guarantees that if `err` is nil, `store` must not be nil.

Comment (tiancaiamao) [inline server/command.go]: this is the same logic as before, along with the comment, I didn't change it.

I agree that `err` is nil and return a `pdpb.Response` with empty content is not Go idiom.
Is here a `errors.Errorf()` ok or we should define some kind of error?

Comment (disksing) [inline server/command.go]: I think remove this comment is enough. @siddontang What do you suggest?

Comment (siddontang) [inline server/command.go]: yep

**Comment by ngaut** [comment]
LGTM

**Pull request merged**